pupil-api/src/services/pupil-authentication.service.ts

```diff
@@ -0,0 +1,92 @@
+'use strict'
+
+import * as azureStorage from 'azure-storage'
+import * as bluebird from 'bluebird'
+import { clone } from 'ramda'
+import * as winston from 'winston'
+
+let azureTableService: any
+const authTable = 'jonPreparedCheck'
+
+export const pupilAuthenticationService = {
+  authenticate: async function authenticate (pupilPin: string, schoolPin: string, tableService?: any): Promise<object> {
+    // set tableService to azureTableService if not provided, but only instantiate it once.
+    if (!tableService) {
+      if (!azureTableService) {
+        azureTableService = azureStorage.createTableService()
+        bluebird.promisifyAll(azureTableService, {
+          promisifier: (originalFunction) => function (...args) {
+            return new Promise((resolve, reject) => {
+              try {
+                originalFunction.call(this, ...args, (error, result, response) => {
+                  if (error) {
+                    return reject(error)
+                  }
+                  resolve({ result, response })
+                })
+              } catch (error) {
+                reject(error)
+              }
+            })
+          }
+        })
+      }
+      tableService = azureTableService
+    }
+
+    const { result } = await tableService.retrieveEntityAsync(authTable, schoolPin, pupilPin)
+    if (result.isCollected._ === true) {
+      throw new Error('Repeat attempts to collect data are not allowed')
```

Comment:
If we bail out here the pupil will only be able to authenticate once and once only.
@GuyHarwood - thoughts?

Reply:
good point.  we need a retry mechanism that is capped.  i will add a card. needs discussion with BAs/PO